### DISCUSSION TITLE
GFDL vortex tracker and gnu

### DISCUSSION THREAD

**Discussion by fipoucat** [dtcenter/METplus#1624] [Installation] [answered]
I need the GFDL Vortex tracker to be associated with MET/Metplus. After many attempts to compile the vortex tracker, I noticed that all compilation options require an Intel compiler I don't have. I finally installed Met without the GFDL Vortex tracker.
I wonder if someone installed the vortex tracker with the GNU (gfortan, gcc) compiler or if someone can help with installing the GFDL vortex tracker with this free compiler.

Thank you

**Comment by georgemccabe** [comment]
Hi @fipoucat,

The GFDL Tracker is an optional external component to METplus that is only used for a few use cases. We typically don't provide support for external components. That said, I modified the vortex tracker code to get it to compile using GNU compilers so we could use it in our automated testing suite. Here is the Dockerfile that is used to obtain and install the tracker and its dependencies:

https://github.com/dtcenter/METplus/blob/main_v4.1/scripts/docker/docker_env/Dockerfile.gfdl-tracker

You may be able to run the commands listed in this file to install the tracker using GNU compilers. The approach is a bit hacky and involves obtaining a tarfile that I created which contains a modified version of the tracker, running some commands, and copying other files in place. I was told that the project that developed the GFDL tracker is no longer funded to work on it, so there is currently no plan to apply the changes needed to support GNU compilers.

I recall from another discussion that you were using newer versions of the GNU compilers to install MET than what we have used to test. It is possible that these versions of the compilers will not be compatible with the modified version of the tracker. If that is the case, we unfortunately will not be able to provide additional installation support since it is outside the scope of the METplus work.

Please let me know how it goes.

**Comment by fipoucat** [comment]
Hi @ <https://github.com/fipoucat>george,

I used the wget command but looks the file is no longer there?

**Reply by georgemccabe** [reply]
Hi @fipoucat, I checked that the file was there yesterday and was able to download it again today.

**Comment by fipoucat** [comment]
OK, it was ssl problem with wget, now got the file.

**Comment by georgemccabe** [comment]
Hi @fipoucat, were you able to resolve this issue?

**Comment by fipoucat** [comment]
***@***.***, I still have problems mainly due to the Makefiles of the
libraries I already installed during the Met installation. I don't need the
vortex tracker to install them again in a separate directory.

**Reply by georgemccabe** [reply]
I'm sorry, I don't follow. Were you able to install the GFDL Tracker using the GNU compilers?

**Comment by fipoucat** [comment]
Hello George, I am facing a problem in installing the GFDL Vortex tracker
because it can't locate the libraries. I think it is related to the
Makefile adjustment for my system.

below is the typical kind of error I am having:

/usr/bin/ld: cannot find -lw3emc_i4r8
/usr/bin/ld: cannot find -lw3nco_i4r8
collect2: error: ld returned 1 exit status
make[2]: [Makefile:12: ../../trk_exec/vint.exe] Error 1 (ignored)
make[2]: Leaving directory
'/home/parallels/METPLUS/standalone_gfdl-vortextracker_v3.9a_gcc/trk_src/vint'
cd tave ; make
make[2]: Entering directory
'/home/parallels/METPLUS/standalone_gfdl-vortextracker_v3.9a_gcc/trk_src/tave'
gfortran -I../../libs/mods/g2  tave.f -L../../libs -lw3emc_i4r8
-lw3nco_i4r8 -lbacio -lg2 -L/home/parallels/MET111/external_libs/lib -lz
-lpng -ljasper -o ../../trk_exec/tave.exe
/usr/bin/ld: cannot find -lw3emc_i4r8
/usr/bin/ld: cannot find -lw3nco_i4r8
collect2: error: ld returned 1 exit status
make[2]: [Makefile:12: ../../trk_exec/tave.exe] Error 1 (ignored)
make[2]: Leaving directory
'/home/parallels/METPLUS/standalone_gfdl-vortextracker_v3.9a_gcc/trk_src/tave'
make[1]: Leaving directory
'/home/parallels/METPLUS/standalone_gfdl-vortextracker_v3.9a_gcc/trk_src'

**Reply by georgemccabe** [reply]
Hi @fipoucat, I'm sorry to hear that you are still having issues installing the GFDL Tracker code. Unfortunately debugging these errors are outside of the scope of the support that the METplus team provides since this is an external software tool. I reached out to my colleagues to ask if there is some sort of help forum for the tracker that I could direct you to, but it appears that funding for that work has ended so there are no means of support available from them.

**Comment by ghost** [comment, marked as answer]
So the issue with GNU 10+ for ubuntu 22.04 has to do with the removal of some silent processing of old fortran code.  On Ubuntu 20.04, which had GNU9, boz files and scalar integers could be calculated even if they were not in modern code format.

With GNU 10+ this changed and the following flags were needed for FCFLAGS and FFLAGS.

-fallow-argument-mismatch
-fallow-invalid-boz

So when trying to install the vortex following package on ubuntu 22.04 these flags would need to be added somewhere.  Sadly the make files are pointing to several locations and nailing down where they go isn't easy.

Perhaps someone who is familiar with the code and make files can figure it out now that you know this flags are needed?

Hope this helps.

-Will
